torchtune/_cli/tune.py

```diff
@@ -4,133 +4,49 @@
 # This source code is licensed under the BSD-style license found in the
 # LICENSE file in the root directory of this source tree.
 
-# Copyright (c) Facebook, Inc. and its affiliates.
-# All rights reserved.
-#
-# This source code is licensed under the BSD-style license found in the
-# LICENSE file in the root directory of this source tree.
-
-"""
-Launcher and utilities for torchtune recipes
-
-``tune`` provides functionality for launching torchtune recipes as well as local
-recipes. Aside from torchtune recipe utilties it integrates with ``torch.distributed.run``
-to support distributed job launching by default. ``tune`` offers everyting that ``torchrun``
-does with the following additional functionalities:
-
-1. ``tune <recipe> <recipe_args>`` with no optional ``torchrun`` options launches a single python process
-
-2. ``<recipe>`` and recipe arg ``<config>`` can both be passed in as names instead of paths if they're included in torchtune
-
-3. ``tune <path/to/recipe.py> <recipe_args>`` can be used to launch local recipes
-
-4. ``tune <torchrun_options> <recipe> <recipe_args>`` will launch a torchrun job
-
-.. note:: ``tune`` is a python
-          `console script <https://packaging.python.org/en/latest/specifications/entry-points/#use-for-scripts>`_
-          to the main module
-          `scripts.cli_utils.tune <https://github.com/pytorch/torchtune/blob/main/scripts/cli_utils/tune>`_
-          declared in the ``scripts`` configuration in
-          `setup.py <https://github.com/pytorch/torchtune/blob/main/setup.py>`_.
-          It is equivalent to invoking ``python -m scripts.cli_utils.tune``.
-"""
 import argparse
-import runpy
-import sys
-from pathlib import Path
 
-import torchtune
-from torch.distributed.run import get_args_parser, run
-from torchtune import list_recipes
-from torchtune._cli import list_scripts
-from torchtune.utils._distributed import _valid_distributed_single_node_nnodes
+from torchtune._cli.cp import Copy
+from torchtune._cli.download import Download
+from torchtune._cli.ls import List
+from torchtune._cli.run import Run
+from torchtune._cli.validate import Validate
 
 
-def _update_parser_help(parser):
-    parser.description = "Torch Tune Recipe Launcher"
-    parser.usage = "tune [options] <recipe> [recipe_args]"
-    parser.formatter_class = argparse.RawDescriptionHelpFormatter
+class TuneCLIParser:
```

Comment:
wonder if we should rename `TuneArgumentParser` as well to make a clearer distinction between these two?

Reply:
Yep, would you like me to add this change to this PR?

Reply:
`TuneRecipeParser`?

Reply:
if you can that would be nice, but it's non-blocking

Reply:
Yeah let's do it separately, no need to fold into this already-massive PR